### DISCUSSION TITLE
Do Images with Captions have to have a specific image name.

### DISCUSSION THREAD

**Discussion by InternalMegaT** [TheLastBen/fast-stable-diffusion#1487] [Q&A]
When using Captions do the images have to be named specifically as well. Is it recommended to name them specifically? I'm training several styles in one model. So, if there not named specifically and I'm training them using captions will it interfere with the other styles and mess everything up?

**Comment by InternalMegaT** [comment]
My previous model used identifiers and classes for each style. Like
**phtoralcsl Photorealistic = Photorealistic
oocsle Oil on Canvas = Oil on Canvas
etc.**
This was my old way of doing it but now I'm trying to get into captions so I can just type what style I want without having to be very specific about the identifiers and having to remember them.

**Reply by TheLastBen** [reply]
if you want to exclude the images filenames from the captions, rename the images into numbers : 1.png ... 10.png ...

**Reply by InternalMegaT** [reply]
Can you make a switch that I can activate that automatically disables filenames so I don't have to go through a 20 thousand image database and rename every caption and image specifically. I genuinely think this would be a good feature.

**Reply by TheLastBen** [reply]
Run this and it will rename all the images (once they are uploaded) to a chosen name: 

```
from subprocess import call
i=0
for filename in tqdm(os.listdir(INSTANCE_DIR), bar_format='  |{bar:15}| {n_fmt}/{total_fmt} Renamed'):
  extension = filename.split(".")[-1]
  identifier=filename.split(".")[0]
  new_path_with_file = os.path.join(INSTANCE_DIR, "choose_a_name_or_a_number"+str(i)+"."+extension)
  call('mv "'+os.path.join(INSTANCE_DIR,filename)+'" "'+new_path_with_file+'"', shell=True)
  i=i+1   
```

**Reply by ProgGrog** [reply]
Apologies for jumping onto a thread, but thoughts re Captions:
I noticed the Caption editor is no longer in the UI.   So, I'm using the Train>Preprocess Images to batch the creation of captions.
1) I like the idea of the renaming option at this step.  Your code is very helpful.
2) It would be very helpful to define a Destination Directory for Captions separate from Destination for Images.  (because Google Drive is a pain for separating txt/png afterwards, since you can't sort by file type / extension in Gdrive, this feature would be super useful.
3) I liked the previous Caption editor, but this entire thing is amazing so I can't complain.  Keep up the great work!  Thanks

**Reply by TheLastBen** [reply]
the captions files are automatically separated from the instance images and put inside the folder named captions, upload the captions and the images together using the instance images cell.

**Reply by InternalMegaT** [reply]
> Run this and it will rename all the images (once they are uploaded) to a chosen name:
> 
> ```
> from subprocess import call
> i=0
> for filename in tqdm(os.listdir(INSTANCE_DIR), bar_format='  |{bar:15}| {n_fmt}/{total_fmt} Renamed'):
>   extension = filename.split(".")[-1]
>   identifier=filename.split(".")[0]
>   new_path_with_file = os.path.join(INSTANCE_DIR, "choose_a_name_or_a_number"+str(i)+"."+extension)
>   call('mv "'+os.path.join(INSTANCE_DIR,filename)+'" "'+new_path_with_file+'"', shell=True)
>   i=i+1   
> ```

 After finally trying the code, It doesn't work I'm keep getting this error:
 ---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
[<ipython-input-1-3dd39c23263c>](https://localhost:8080/#) in <module>
      1 from subprocess import call
      2 i=0
----> 3 for filename in tqdm(os.listdir(INSTANCE_DIR), bar_format='  |{bar:15}| {n_fmt}/{total_fmt} Renamed'):
      4   extension = filename.split(".")[-1]
      5   identifier=filename.split(".")[0]

NameError: name 'tqdm' is not defined

Even after fixing that error I still can't get it find the location of the folder in my google drive. I try codes like this
```
import os
import tqdm
from google.colab import drive
from googleapiclient.discovery import build

drive.mount('/content/drive')

directory_path = "/content/drive/MyDrive/location"
if not os.path.exists(directory_path):
    os.makedirs(directory_path)

service = build('drive', 'v3')


folder_id = "<folder_id>"
results = service.files().list(q=f"mimeType='image/jpeg' and trashed = false and parents in '{folder_id}'", 
                               fields="nextPageToken, files(id, name)").execute()
items = results.get("files", [])

i = 0
for item in tqdm(items, bar_format='  |{bar:15}| {n_fmt}/{total_fmt} Renamed'):
    file_id = item['id']
    filename = item['name']
    extension = filename.split(".")[-1]
    new_filename = "choose_a_name_or_a_number" + str(i) + "." + extension
    service.files().update(fileId=file_id, body={'name': new_filename}).execute()
    i += 1
```
    
And I still have been getting the same can't find location error. I want the code to rename files stored in my captions and instance_images folder to the appropriate names to solve this issue. I can't figure out how to do that. Also keep in mind that I want it to rename the captions the same as the paired images are.

**Reply by ProgGrog** [reply]
Not to distract, but I know Windows likes to use (x) as the increment suffix, which I believe is a bad habit, because while _1, _2, is handled fine in Dreambooth, where the _ and #'s are removed, the round parenthesis bracket is (#) is not.  I'd suggest that a renaming formula remove (#) and replace with _#_ as part of any algorithm.  I'm not qualified to suggest the correct python code though.

Relatedly, in two other discussions I'd found alternatives suggested for renaming/resizing large batches on Windows.  They both seem to work well.
FastStone Photo Resizer
Cupscale
Neither of these can help with caption txt files or txt file renaming, unfortunately.

**Comment by TheLastBen** [comment]
add `from tqdm import tqdm` at the begining
